azkaban-web-server/src/main/java/cloudflow/daos/ProjectDaoImpl.java

```diff
@@ -0,0 +1,164 @@
+  private static final Logger log = LoggerFactory.getLogger(ProjectDaoImpl.class);
+
+  static String INSERT_PROJECT =
+      "INSERT INTO project (name, description, space_id, created_by, creation_time, "
+          + "last_modified_by, last_modified_time, version,  active, enc_type, settings_blob) "
+          + "values "
+          + "(?,?,?,?,?,?,?,?,?,?,?)";
+
+  @Inject
+  public ProjectDaoImpl(DatabaseOperator dbOperator, ProjectAdminDaoImpl projectAdminDaoImpl) {
+    this.dbOperator = dbOperator;
+    this.projectAdminDaoImpl = projectAdminDaoImpl;
+  }
+
+  @Override
+  public String create(Project project, User user) {
+    String name = project.getName();
+    FetchProjectHandler fetchProjectHandler = new FetchProjectHandler();
+
+    // Check if the same project name exists.
+    try {
+      final List<Project> projects = this.dbOperator
+          .query(FetchProjectHandler.SELECT_PROJECT_WITH_NAME_AND_SPACE_ID, fetchProjectHandler,
+              name, project.getSpaceId());
+      if (!projects.isEmpty()) {
+        throw new ProjectManagerException(
+            "Project with name " + name + " already exists.");
+      }
+    } catch (final SQLException ex) {
+      log.error("Check if project '" + name + "' exists failed with: " + ex);
+      throw new ProjectManagerException("Check if project '" + name + "' exists failed");
+    }
+
+    final SQLTransaction<Long> insertAndGetProjectId = transOperator -> {
+      String currentTime = DateTime.now().toLocalDateTime().toString();
+      transOperator.update(INSERT_PROJECT, project.getName(), project.getDescription(),
+          project.getSpaceId(), user.getUserId(), currentTime, user.getUserId(), currentTime,
+          1, true, 2, null);
+      transOperator.getConnection().commit();
+      return transOperator.getLastInsertId();
+    };
+
+    String projectId = "";
+    try {
+      projectId = dbOperator.transaction(insertAndGetProjectId).toString();
+      if (projectId.isEmpty()) {
+        throw new ProjectManagerException("Insert project for '" + name + "' failed.");
+      }
+      projectAdminDaoImpl.addAdmins(projectId, project.getAdmins());
+    } catch (SQLException ex) {
+      log.error(INSERT_PROJECT + " failed with exception:", ex);
+      throw new ProjectManagerException("Insert project for '" + name + "' failed.",
+          ex);
+    }
+    return projectId;
+  }
+
+  @Override
+  public Optional<Project> get(String projectId) {
+    List<Project> projects = new ArrayList<>();
+    FetchProjectHandler fetchProjectHandler = new FetchProjectHandler();
+    try {
+      projects = dbOperator.query(FetchProjectHandler.SELECT_PROJECT_WITH_ID, fetchProjectHandler,
+          projectId);
+      for(Project project : projects) {
+        project.setAdmins(projectAdminDaoImpl.findAdminsByProjectId(project.getId()));
+      }
+    } catch (SQLException ex) {
+      log.error("Select for project record with id " + projectId + " failed: ", ex);
```

Comment:
I think SQLExceptions should bubble up as a 500 response not 404 which is currently the case.

Reply:
I agree with this. We need to decide on exception handling strategy along with the errors to return, Let's do that today and I will make changes based on it.